### ISSUE TITLE
Limited, but official Tweakscale support vs none.

### ISSUE THREAD

**Issue by stratochief66** [KSP-RO/RealismOverhaul#1624]
Many people are of the opinion that Tweakscale still serves a valuable service in RO, providing the ability to customize structural parts. The challenge is to accomplish that without leading people to use it for other things, like tanks or aero parts or anything other than structural/filler parts.

In particular, TweakScale parts behave terribly with respect to the aero system (FAR). @ferram4, was I dreaming when you suggested that a recent change to FAR may make this better, or make FAR work well for SSTU parts?

Oooor, just no support at all, and people can use it for structural parts on their own, in secret?

**Comment by raidernick** [comment]
i vote no support ever

**Comment by NathanKell** [comment]
Does IR still require it?

**Comment by raidernick** [comment]
I wasn't aware IR even worked still, i don't know of any mods that still use it.

**Comment by pap1723** [comment]
FWIW, I vote for limited support. For example, I do not make planes of any kind, so in order to try and keep my folder manageable, I do not install B9. When I am launching my early Sounding Rockets, I need to  have a fin that I can TweakScale down to fit the size of the rocket. However, that is the only use I have for it, so maybe my answer is I just make a duplicate of the Basic stock fin and include the CFG of it for all players?

**Comment by NathanKell** [comment]
(a) Please never ever tweakscale wings. That's why RP-0 requires procedural wings. Besides which, using tweakscaled wings breaks the temperature-based tech progression of wings.

(b) I vote for limited tweakscale support--we don't have procedural girders, AFAIK, so there's still a solid use case.

**Comment by SirKeplan** [comment]
Until someone makes procedural girders, panels, landing legs, etc. we should keep limited tweakscale support imho.

**Comment by assassinacc** [comment]
I think limited support is a good middle ground.
Personally I like to use it for IR, but nothing else.

**Comment by Bornholio** [comment]
I use it for structural and things like nathankell mentions.  We need to make it clear that Tweakscaled tanks can cause negative mass problems with RF interactions.  I wish we had a better implementation that didn't make small parts floppy and manipulate OrigMass.

**Comment by pap1723** [comment]
@Bornholio and anyone else who has experience with this. Should we eliminate all TweakScale modules on all parts except for Structural Parts? Is there anything else that makes sense to have it available for?

**Comment by siimav** [comment]
I also vote for limited support. Personally I have used it for structural parts, nosecones, intakes and solar panels. I think Tweakscale is almost a requirement for making a space station that resembles the ISS.

**Comment by pap1723** [comment]
Solar Panels huh? Do they scale correctly?

**Comment by siimav** [comment]
I think so. Was using it to upscale the solar panels for my space station because the prefab ones are way too small for an ISS-lookalike.

**Comment by kurgut** [comment]
@pap1723 they seem to do. 
But the actual "tweaked" scale doesn't match with the one in the GUI.
eg you scale *2, but the panel is effectively scaled by 4 (something like that, but more than 2) in the VAB/ingame.

**Comment by Bornholio** [comment]
solar does see to function properly, or atleast 1.3.1 and earlier it has, have not tried it post
They are area (^2) function vs Linear % size as they should be.  I'm okay with just removing support on anything that gets RF basemass modified, maybe also fins/wings if FASA support is damaged.

My experience is that a small tank can go negative mass and thus generate anchoring effect or false forces.  Also once had a solar panel that base clipped into structure and generated false forces when un folded.  I suspect the second problem was more to do with Vens having a wacky normal map/.mu

**Comment by DRVeyl** [comment]
Couple years later.
It seems the only functional use for Tweakscale is still in structural parts like girders, where there's really nothing interesting going on and we're just scaling a shaped lump of metal.
Solar panels rescaling and tech progression is well handled through ROSolar, along with its access to multiple models, including models provided by other parts packs [once ROSolar configs are written].

We've started to run into issues with KSP Recall, a dependency of Tweakscale and by the same current mod maintainer, causing, well, issues that get reported to us.  (The current common one is it mis-placing the attachment nodes on a ProcParts root part, for which ProcParts handles this exactly as it should).

Could write a simple system looking for a flag in the part ie "#roKeepTweakscale" and then a patch in :LAST that removes Tweakscale from everything without the flag.  That lets us add the flag for a few chosen parts and then mass-remove it [and not have to write many hundreds of patches to eliminate it, much like what I just did for handling reaction wheels.

Thoughts?

**Comment by NathanKell** [comment]
Structurals *and intakes* fwiw.
Honestly we might be better served by making a stripped down rescaler from
the ROLib codebase and conflicting TS.

Sent by my thumbs, slowly.

**Comment by siimav** [comment]
> Honestly we might be better served by making a stripped down rescaler from
the ROLib codebase and conflicting TS.

Agreed.

**Comment by siimav** [comment]
There's TS/R now which comes with very little bloat and doesn't outright break RO. Also some attempts have been made to exclude TS/R scaling support from parts that outright wouldn't work (like all the parts that are already procedural). I haven't done any further analysis on whether there are other parts where scaling support or just the cost/mass factors make no sense.
IMO TS/L should definitely be added as a conflict to prevent players unknowingly installing the inferior version.

**Comment by Clayell** [comment]
Just as quick note, has anyone looked into TweakScale Rescaled Redistributable? (as opposed to TweakScale Redistributable, which is the one that FAR requires) It seems to be a new version that JonnyOThan has been contributing to. Should it (and the corresponding TweakScale Rescaled) be marked as conflicting as well?

**Comment by siimav** [comment]
No, we can't mark both redistributables as conflicting. However marking TS/L and its associated redistributable as a conflict would remove that choice entirely.

**Comment by Clayell** [comment]
TS/L?
